mesonbuild/compilers/fortran.py

```diff
@@ -0,0 +1,281 @@
+
+    def get_depfile_suffix(self):
+        return 'd'
+
+    def get_std_exe_link_args(self):
+        return []
+
+    def build_rpath_args(self, build_dir, rpath_paths, install_rpath):
+        return build_unix_rpath_args(build_dir, rpath_paths, install_rpath)
+
+    def module_name_to_filename(self, module_name):
+        return module_name.lower() + '.mod'
+
+    def get_warn_args(self, level):
+        return ['-Wall']
+
+    def get_no_warn_args(self):
+        return ['-w']
+
+
+class GnuFortranCompiler(FortranCompiler):
+    def __init__(self, exelist, version, gcc_type, is_cross, exe_wrapper=None, defines=None):
+        super().__init__(exelist, version, is_cross, exe_wrapper=None)
+        self.gcc_type = gcc_type
+        self.defines = defines or {}
+        self.id = 'gcc'
+
+    def has_builtin_define(self, define):
+        return define in self.defines
+
+    def get_builtin_define(self, define):
+        if define in self.defines:
+            return self.defines[define]
+
+    def get_always_args(self):
+        return ['-pipe']
+
+    def gen_import_library_args(self, implibname):
+        """
+        The name of the outputted import library
+
+        Used only on Windows
+        """
+        return ['-Wl,--out-implib=' + implibname]
+
+
+class G95FortranCompiler(FortranCompiler):
+    def __init__(self, exelist, version, is_cross, exe_wrapper=None):
+        super().__init__(exelist, version, is_cross, exe_wrapper=None)
+        self.id = 'g95'
+
+    def get_module_outdir_args(self, path):
+        return ['-fmod=' + path]
+
+    def get_always_args(self):
+        return ['-pipe']
+
+    def get_no_warn_args(self):
+        # FIXME: Confirm that there's no compiler option to disable all warnings
+        return []
+
+    def gen_import_library_args(self, implibname):
+        """
+        The name of the outputted import library
+
+        Used only on Windows
+        """
+        return ['-Wl,--out-implib=' + implibname]
+
+
+class SunFortranCompiler(FortranCompiler):
+    def __init__(self, exelist, version, is_cross, exe_wrapper=None):
+        super().__init__(exelist, version, is_cross, exe_wrapper=None)
+        self.id = 'sun'
+
+    def get_dependency_gen_args(self, outtarget, outfile):
+        return ['-fpp']
+
+    def get_always_args(self):
+        return []
+
+    def get_warn_args(self, level):
+        return []
+
+    def get_module_outdir_args(self, path):
+        return ['-moddir=' + path]
+
+
+class IntelFortranCompiler(IntelCompiler, FortranCompiler):
+    std_warn_args = ['-warn', 'all']
+
+    def __init__(self, exelist, version, is_cross, exe_wrapper=None):
+        self.file_suffixes = ('f90', 'f', 'for', 'ftn', 'fpp')
+        FortranCompiler.__init__(self, exelist, version, is_cross, exe_wrapper)
+        # FIXME: Add support for OS X and Windows in detect_fortran_compiler so
+        # we are sent the type of compiler
+        IntelCompiler.__init__(self, ICC_STANDARD)
```

Comment:
__[flake8]__
* [F405] 'ICC_STANDARD' may be undefined, or defined from star imports: .compilers

<!-- Comment by [SideCI](https://sideci.com) -->